src/main/java/org/jabref/gui/externalfiletype/EditExternalFileTypeEntryDialog.java

```diff
@@ -62,6 +62,7 @@ public void initialize() {
         icon.setGraphic(viewModel.getIcon());
 
         defaultApplication.selectedProperty().bindBidirectional(viewModel.defaultApplicationSelectedProperty());
+        customApplication.selectedProperty().bindBidirectional(viewModel.customApplicationSelectedProperty());
```

Comment:
Custom application selected is always the opposite of default application, right?
I guess the problem is that the toggle group is defined locally
https://github.com/JabRef/jabref/blob/d0393b59b96c09bdf53cfc053845764764815a44/src/main/java/org/jabref/gui/externalfiletype/EditExternalFileTypeEntryDialog.fxml#L38
and not globally as in
https://github.com/JabRef/jabref/blob/d0393b59b96c09bdf53cfc053845764764815a44/src/main/java/org/jabref/gui/preferences/EntryEditorTab.fxml#L14-L17

If this does not work, then you can also try
```
customApplication.selectedProperty().bind(viewModel.defaultApplicationSelectedProperty().not());
```

Reply:
Yes, the radio button group should toggle and only one radio button should be active. First I tried `customApplication.selectedProperty().bindBidirectional(defaultApplicationSelectedProperty().not());` which is of course not possible. On the other hand, I needed a bidirectional binding, because the radio button can be changed from both directions (program or user), thus I implemented it this way. (Which works.)

Now I have tried your suggestion with `customApplication.selectedProperty().bind(viewModel.defaultApplicationSelectedProperty().not());` (so a onedirectional binding) out of curiosity, but, as expected, issues resulted. In this case, when I try to switch to the "custom" radio button, then an exception is thrown, stating `java.lang.RuntimeException: RadioButton.selected : A bound value cannot be set.`, since now the value can only be set by the program but not by the user.

In the end, I looked how it was done in

```
 <fx:define> 
     <ToggleGroup fx:id="nameFormat"/> 
     <ToggleGroup fx:id="firstNames"/> 
 </fx:define> 
```

Here, also a bidirectional mapping for all radio buttons gets declared:

        autoCompleteFirstLast.selectedProperty().bindBidirectional(viewModel.autoCompleteFirstLastProperty());
        autoCompleteLastFirst.selectedProperty().bindBidirectional(viewModel.autoCompleteLastFirstProperty());
        autoCompleteBoth.selectedProperty().bindBidirectional(viewModel.autoCompleteBothProperty());

But only the `true` value is getting set, which is of course sufficient:

        if (autoCompletePreferences.getOnlyCompleteFirstLast()) {
            autoCompleteFirstLastProperty.setValue(true);
        } else if (autoCompletePreferences.getOnlyCompleteLastFirst()) {
            autoCompleteLastFirstProperty.setValue(true);
        } else {
            autoCompleteBothProperty.setValue(true);
        }

And so did I now.

Reply:
Radio button in a toggle group dont support bidirectional binding.
I stumbled across this problem already.
You always need to work around that. 
Just look how it's done in the settings dialog

Reply:
I made it as described above and it is coded this way in the settings dialogue as well (see above) and it works. Is this ok now?